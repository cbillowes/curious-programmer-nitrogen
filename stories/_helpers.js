import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page