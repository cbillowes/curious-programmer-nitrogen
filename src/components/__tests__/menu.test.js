import React from 'react'
import Menu from '../menu'
import { getSnapshot } from './_helpers'

describe(`Menu`, () => {
  it(`should render`, () => {
    const menu = (
      <Menu
        toggleOnClick={jest.fn()}
      />
    )
    const tree = getSnapshot(menu)
    expect(tree).toMatchSnapshot()
  })

  it(`should render opened`, () => {
    const menu = (
      <Menu
        toggleOnClick={jest.fn()}
        isOpen={true}
      />
    )
    const tree = getSnapshot(menu)
    expect(tree).toMatchSnapshot()
  })

  it(`should render closed`, () => {
    const menu = (
      <Menu
        toggleOnClick={jest.fn()}
        isOpen={false}
      />
    )
    const tree = getSnapshot(menu)
    expect(tree).toMatchSnapshot()
  })
})
