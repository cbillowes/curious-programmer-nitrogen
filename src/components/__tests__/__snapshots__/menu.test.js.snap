// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render 1`] = `
<button
  className="menu-button"
  onClick={[MockFunction]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-ellipsis-v fa-w-6 "
    data-icon="ellipsis-v"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 192 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`Menu should render closed 1`] = `
<button
  className="menu-button"
  onClick={[MockFunction]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-ellipsis-v fa-w-6 "
    data-icon="ellipsis-v"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 192 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`Menu should render opened 1`] = `
<button
  className="menu-button active"
  onClick={[MockFunction]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-ellipsis-v fa-w-6 "
    data-icon="ellipsis-v"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 192 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
